lib/matrix.ex

```diff
@@ -53,6 +54,43 @@ defmodule Matrix do
     Result.error("It is not possible create square matrix with negative row or column!")
   end
 
+  @doc """
+  Addition of two matrices. Sizes (dimensions) of both matrices must be same.
+  Otherwise you get error message.
+
+  Returns result, it means either tuple of {:ok, matrix} or {:error, "msg"}.
+
+  ## Examples
+
+      iex> mat1 = {:ok, [[1, 2, 3], [4, 5, 6], [7, 8, 9]]}
+      iex> mat2 = Matrix.new(3,1)
+      iex> Matrix.and_then2(mat1, mat2, &Matrix.add(&1, &2))
```

Comment:
Tady uz by to asi chtelo mit nejakou tu fci v Resultu `and_thenX` jak jsi psal. Neb ja ji tu pak vyuziji casteji i pro dalsi maticove operace apod...